clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -160,92 +228,91 @@ private boolean maybeTransitionToStable() {
+            throw new IllegalStateException("A target assignment pending to be reconciled already" +
+                    " exists.");
         }
     }
 
-    private boolean hasPendingTargetAssignment() {
-        return targetAssignment.isPresent() || nextTargetAssignment.isPresent();
-    }
-
-
-    /**
-     * Update state and assignment as the member has successfully processed a new target
-     * assignment.
-     * This indicates the end of the reconciliation phase for the member, and makes the target
-     * assignment the new current assignment.
-     *
-     * @param assignment Target assignment the member was able to successfully process
-     */
-    public void onAssignmentProcessSuccess(ConsumerGroupHeartbeatResponseData.Assignment assignment) {
-        updateAssignment(assignment);
-        transitionTo(MemberState.STABLE);
-    }
-
     /**
-     * Update state and member info as the member was not able to process the assignment, due to
-     * errors in the execution of the user-provided callbacks.
-     *
-     * @param error Exception found during the execution of the user-provided callbacks
+     * Returns true if the member has a target assignment being processed.
      */
-    public void onAssignmentProcessFailure(Throwable error) {
-        transitionTo(MemberState.FAILED);
-        // TODO: update member info appropriately, to clear up whatever shouldn't be kept in
-        //  this unrecoverable state
+    private boolean hasPendingTargetAssignment() {
+        return targetAssignment.isPresent();
     }
 
     private void resetEpoch() {
         this.memberEpoch = 0;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public MemberState state() {
         return state;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public AssignorSelection assignorSelection() {
         return this.assignorSelection;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public ConsumerGroupHeartbeatResponseData.Assignment assignment() {
+    public ConsumerGroupHeartbeatResponseData.Assignment currentAssignment() {
         return this.currentAssignment;
     }
 
+
+    /**
+     * Assignment that the member received from the server but hasn't completely processed yet.
+     */
     // VisibleForTesting
     Optional<ConsumerGroupHeartbeatResponseData.Assignment> targetAssignment() {
         return targetAssignment;
     }
 
-    // VisibleForTesting
-    Optional<ConsumerGroupHeartbeatResponseData.Assignment> nextTargetAssignment() {
-        return nextTargetAssignment;
-    }
-
     /**
-     * Set the current assignment for the member. This indicates that the reconciliation of the
-     * target assignment has been successfully completed.
-     * This will clear the {@link #targetAssignment}, and take on the
-     * {@link #nextTargetAssignment} if any.
+     * This indicates that the reconciliation of the target assignment has been successfully
```

Comment:
instead of "this indicates" can we say: "this is invoked when.... is successfully completed"?

Reply:
When and how this logic is invoked is changing in the [reconciler PR ](https://github.com/apache/kafka/pull/14357) after our sync for reviewing how the interactions between HB-reconciler-manager, so I would suggest we wait until we settle on that PR and adjust there according to the changes, makes sense?